README.md

```diff
@@ -1,93 +1,48 @@
 
-When you submit a pull request, a CLA-bot will automatically determine whether you need to provide
-a CLA and decorate the PR appropriately (e.g., label, comment). Simply follow the instructions
-provided by the bot. You will only need to do this once across all repos using our CLA.
+## Auto-generate your package in TypeScript using Autorest
 
-This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
-For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
-contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
-
-# Simple AutoRest Extension
+It is easy to generate an SDK once you have a swagger specification file.
 
-A simple AutoRest extension that can be used as a template for or documentation about writing such extensions.
-Since this a TypeScript project, we distinguish between language agnostic and language dependent requirements.
+- You will need first to install Autorest
 
-## Language Agnostic Requirements
-
-More information about these can be found [here](https://github.com/Azure/autorest/tree/master/docs/developer), summary:
-
-1. Specify the start command of the extension as the `start` script in the `package.json`
-2. Implement the AutoRest extension protocol (here: handled by library, see below)
-3. Hook up plugins into the AutoRest pipeline DAG, e.g.
+```bash
+npm install -g autorest
+```
 
-```yaml
-version: 3.1.2
-use-extension:
-  "@autorest/modelerfour": "4.15.456"
+- You can then generate the SDK as follows:
 
-modelerfour:
-  # this runs a pre-namer step to clean up names
-  prenamer: true
-  # this will flatten modelers marked with 'x-ms-client-flatten'
-  flatten-models: true
-  # this will flatten parameters marked with 'x-ms-client-flatten'
-  flatten-payloads: true
-  # this will make the content-type parameter always specified
-  always-create-content-type-parameter: true
-  # enables parameter grouping via x-ms-parameter-grouping
-  group-parameters: true
+```bash
+autorest --typescript <path to the swagger file>
+```
 
-pipeline:
-  typescript: # <- name of plugin
-    input: modelerfour/identity
-    output-artifact: typescript-files
+- You will likely need to specify extra flags to control the behavior of the generation, and these flags are listed in the next section.
 
-  typescript/emitter:
-    input: typescript
-    scope: typescript-scope/emitter
+## Options
 
-typescript-scope/emitter:
-  input-artifact: typescript-files
+In addition to the [list of Autorest flags](https://github.com/Azure/autorest/blob/master/docs/generate/flags.md), you can further control the behavior of the typescript generator with the following flags:
 
-output-artifact: typescript-files
-```
+| Flag                            | Description                                                                                                                             |

```

Comment:
Can we have the description include a note on when the user would probably want to use it?

For example, the flag for core v2 can point to https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/README.md so that reader gets more information on what they are opting into

--hide-client is probably useful if one is planning to write a convenience layer over the generated code rather than use the generated code directly

etc